src/callbacks_step/amr.jl

```diff
@@ -445,6 +445,66 @@ function (amr_callback::AMRCallback)(u_ode::AbstractVector, mesh::P4estMesh,
   return has_changed
 end
 
+function (amr_callback::AMRCallback)(u_ode::AbstractVector, mesh::T8codeMesh,
+                                     equations, dg::DG, cache, semi,
+                                     t, iter;
+                                     only_refine=false, only_coarsen=false,
+                                     passive_args=())
+
+  has_changed = false
+
+  @unpack controller, adaptor = amr_callback
+
+  u = wrap_array(u_ode, mesh, equations, dg, cache)
+  indicators = @trixi_timeit timer() "indicator" controller(u, mesh, equations, dg, cache, t=t, iter=iter)
+
+  if only_coarsen
+    indicators[indicators .> 0] .= 0
+  end
+
+  if only_refine
+    indicators[indicators .< 0] .= 0
+  end
+
+  @boundscheck begin
+    @assert axes(indicators) == (Base.OneTo(ncells(mesh)),) (
+      "Indicator array (axes = $(axes(indicators))) and mesh cells (axes = $(Base.OneTo(ncells(mesh)))) have different axes"
+    )
+  end
+
+  @trixi_timeit timer() "adapt" begin
+    difference = @trixi_timeit timer() "mesh" trixi_t8_adapt!(mesh,indicators)
+
+    @trixi_timeit timer() "solver" adapt!(u_ode, adaptor, mesh, equations, dg, cache, difference)
+  end
+
+  # Store whether there were any cells coarsened or refined and perform load balancing.
+  has_changed = any(difference .!= 0)
+
+  # TODO: T8codeMesh for MPI not implemented yet.
```

Comment:
Do we have an issue to track the progress of the t8code integration into Trixi.jl? If not, please create it and add TODO items like these to the list.